presidio-structured/presidio_structured/data/data_processors.py

```diff
@@ -0,0 +1,231 @@
+        """
+        Recursively retrieves the value from nested data using a given path.
+
+        :param data: Nested data (list or dictionary).
+        :param path: List of keys/indexes representing the path.
+        :return: Retrieved value.
+        """
+        for i, key in enumerate(path):
+            if isinstance(data, list):
+                if key.isdigit():
+                    data = data[int(key)]
+                else:
+                    return [
+                        JsonDataProcessor._get_nested_value(item, path[i:])
+                        for item in data
+                    ]
+            elif isinstance(data, dict):
+                data = data.get(key)
+            else:
+                return data
+        return data
+
+    @staticmethod
+    def _set_nested_value(
+        data: Union[Dict, List], path: List[str], value: Any
+    ) -> None:
+        """
+        Recursively sets a value in nested data using a given path.
+
+        :param data: Nested data (JSON-like).
+        :param path: List of keys/indexes representing the path.
+        :param value: Value to be set.
+        """
+        for i, key in enumerate(path):
+            if isinstance(data, list):
+                if i + 1 < len(path) and path[i + 1].isdigit():
+                    idx = int(path[i + 1])
+                    while len(data) <= idx:
+                        data.append({})
+                    data = data[idx]
+                    continue
+                else:
+                    for item in data:
+                        JsonDataProcessor._set_nested_value(
+                            item, path[i:], value
+                        )
+                    return
+            elif isinstance(data, dict):
+                if i == len(path) - 1:
+                    data[key] = value
+                else:
+                    data = data.setdefault(key, {})
+
+    def _process(
+        self,
+        data: Union[Dict, List],
+        key_to_operator_mapping: Dict[str, Callable],
+    ) -> Union[Dict, List]:
+        """
+        Operates on the given JSON-like data based on the provided configuration.
+
+        :param data: JSON-like data to be operated on.
+        :param config: Configuration object containing operator information.
+        :return: JSON-like data after the operation.
+        """
+
+        if not isinstance(data, (dict, list)):
+            raise ValueError("Data must be a JSON-like object")
+
+        for key, operator_callable in key_to_operator_mapping.items():
+            self.logger.debug(f"Operating on key {key}")
+            keys = key.split(".")
+            if isinstance(data, list):
+                for idx, item in enumerate(data):
+                    self._process(item, key_to_operator_mapping)
+            else:
+                text_to_operate_on = self._get_nested_value(data, keys)
+                if text_to_operate_on is not None:
```

Comment:
```suggestion
                if not text_to_operate_on:
```

Reply:
I assume you mean if text_to_operate_on (otherwise it negates the existing statement), but agreed.